sm2/src/pke.rs

```diff
@@ -1,73 +1,52 @@
 //! SM2 Encryption Algorithm (SM2) as defined in [draft-shen-sm2-ecdsa § 5].
 //!
 //! ## Usage
-//!
-//! NOTE: requires the `sm3` crate for digest functions and the `primeorder` crate for prime field operations.
-//!
-//! The `DecryptingKey` struct is used for decrypting messages that were encrypted using the SM2 encryption algorithm.
-//! It is initialized with a `SecretKey` or a non-zero scalar value and can decrypt ciphertexts using the specified decryption mode.
-#![cfg_attr(feature = "std", doc = "```")]
-#![cfg_attr(not(feature = "std"), doc = "```ignore")]
-//! # fn example() -> Result<(), Box<dyn std::error::Error>> {
-//! use rand_core::OsRng; // requires 'os_rng` feature
-//! use sm2::{
-//!     pke::{EncryptingKey, Mode},
-//!     {SecretKey, PublicKey}
-//! };
-//!
+#![cfg_attr(feature = "alloc", doc = "```")]
+#![cfg_attr(not(feature = "alloc"), doc = "```ignore")]
+//! use sm2::pke::{EcDecrypt, EcEncrypt, Cipher, Mode};
+//! use sm2::SecretKey;
+//! use rand_core::OsRng;
 //! // Encrypting
-//! let secret_key = SecretKey::try_from_rng(&mut OsRng).unwrap(); // serialize with `::to_bytes()`
+//! let secret_key = SecretKey::try_from_rng(&mut OsRng).unwrap();
```

Comment:
Would prefer the original comment be retained:

```suggestion
//! let secret_key = SecretKey::try_from_rng(&mut OsRng).unwrap(); // serialize with `::to_bytes()`
```